### PR TITLE
Adding CentOS 7 Vagrantfile & build script

### PR DESCRIPTION
**Pull request by l1x**
I have updated the Vagrantfile to have an entry for CentOS 7. Also created a simple build script which is pretty similar to the one in Beringei.

How to test:
```
vagrant up centos7
```
Todo:

Implement -j X for the build.

### CODE REVIEW

File: setup_centos7.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
```
Comment (siying) [inline setup_centos7.sh]: I didn't look into the script but I think scripts like this should be under build_tools/

**Comment by l1x** [comment]
@ajkr not sure what this means. You accept pull requests here and move the content internal first and than later in the next release it will be published? Thanks anyways.

**Comment by ajkr** [comment]
yes it goes internal first. But then it's immediately synced to master by a bot that makes all our commits. So yours is already on master branch here: https://github.com/facebook/rocksdb/commit/d6336563e58e1d8f421862b75b18ffcebd156467.

**Comment by l1x** [comment]
Thanks @ajkr!

**Pull request closed without merging**